src/sage/modules/with_basis/representation.py

```diff
@@ -201,9 +217,234 @@ def twisted_invariant_module(self, chi, G=None, **kwargs):
 
         return super().twisted_invariant_module(G, chi, side=side, **kwargs)
 
+    def representation_matrix(self, g, side=None, sparse=False):
```

Comment:
consider allowing other matrix constructor keywords, in particular `implementation`

Reply:
We can't do this because `matrix.zero` does not support this.

Reply:
Why use that method anyway, not the `matrix` constructor?
I'd also suggest to return an immutable matrix in this method.

Reply:
> Why use that method anyway, not the `matrix` constructor?

`matrix` also does not support that. When I tested this, it seemed faster to manipulate the entries of the matrix rather than feed the data off (more transient objects as well), even more so when the matrices have less support (which can often happen).

> I'd also suggest to return an immutable matrix in this method.

Why? It isn't cached, and it is easy to make matrices immutable if needed (but less so the other way around).

Reply:
> `matrix` also does not support that.

You are right, sorry.

It does support the `space` keyword through which one can prescribe an `implementation`.

Reply:
Anyway, just a wishlist item.